plugins/modules/files/toml_file.py

```diff
@@ -0,0 +1,419 @@
+# In the original values, 5 is already present in C.D.
+# There is no change.
+# Get the following content in /my/dest/file.toml :
+# A = 1
+# B = 2
+#
+# [C]
+# D = 3
+# E = [ 4, 5, 6,]
+
+- name: "Ensure values are present in a list."
+  community.general.toml_file:
+    path: "/my/dest/file.toml"
+    value:
+      C:
+       E:
+        - 7
+    state: "present"
+# In the original values, 7 is not present in C.D, it will be added.
+# There is a change.
+# Get the following content in /my/dest/file.toml :
+# A = 1
+# B = 2
+#
+# [C]
+# D = 3
+# E = [ 4, 5, 6, 7,]
+
+- name: "Ensure values are absent in a list."
+  community.general.toml_file:
+    path: "/my/dest/file.toml"
+    value:
+      C:
+       E:
+        - 6
+    state: "absent"
+# In the orginal values, 6 is present in C.D, it will be removed.
+# There is a change.
+# Get the following content in /my/dest/file.toml :
+# A = 1
+# B = 2
+#
+# [C]
+# D = 3
+# E = [ 4, 5,]
+
+- name: "Ensure values are present depending their index in a list."
+  community.general.toml_file:
+    path: "/my/dest/file.toml"
+    list_diff_type: index
+    value:
+      C:
+       E:
+        -
+        - 7
+    state: "present"
+# In the original value, the second index for C.E list is not 7, values will
+# be updated.
+# Note, if you do not want to change the value in on some check you can set it
+# to none like the first index of C.E in the previous sample.
+# There is a change.
+# Get the following content in /my/dest/file.toml :
+# A = 1
+# B = 2
+#
+# [C]
+# D = 3
+# E = [ 4, 7, 6,]
+
+- name: "Ensure values are absent depending their index in a list."
+  community.general.toml_file:
+    path: "/my/dest/file.toml"
+    list_diff_type: index
+    value:
+      C:
+       E:
+        -
+        - 7
+        - 6
+    state: "absent"
+# In the original datas, the second value of C.E list is not 7 so it will be
+# keept, but the third value is 6, so it will be removed.
+# Note, if you do not want to change the value in on some check you can set it
+# to none like the first index of C.E in the previous sample.
+# There is a change.
+# Get the following content in /my/dest/file.toml :
+# A = 1
+# B = 2
+#
+# [C]
+# D = 3
+# E = [ 4, 5,]
+'''
+
+RETURN = r'''
+  changed:
+    description:
+      - Indicate whether the TOML file is changed or not.
+    returned: always
+    type: bool
+  created:
+    description:
+      - Indicate if the file TOML is created or not.
+    returned: success
+    type: bool
+  diff:
+    description:
+      - Contain a dict with two elements, C(after) contain the value after
+        the change and C(before) contain values before the change.
+    returned: changed
+    type: dict
+  failed:
+    description:
+      - Indicate if the module has failed or not.
+    returned: always
+    type: bool
+  result:
+    description:
+      - Contain the values that are finally present in the TOML file.
+    returned: success
+    type: dict
+'''
+
+
+import toml
```

Comment:
This import must be guarded. See the explanation: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/import.html